typing/printtyp.ml

```diff
@@ -1959,6 +1995,8 @@ let may_prepare_expansion compact (t, t') =
       mark_loops t; (t, t)
   | _ -> prepare_expansion (t, t')
 
+let print_path f p = !Oprint.out_ident f (tree_of_path Type p)
```

Comment:
Note: this function is much more difficult to use correctly that it appears: the construction of `tree_of_path` is effectful and mutate the global state of `short-path` and the naming context. It is quite often not the right thing to delay the construction of the outcome tree to the printing time.

An easier function to use correctly would be
```ocaml
let print_path p = Format.dprintf "%a" !Oprint.out_ident (tree_of_path Type p)
```
This variant build the tree immediately (in the current context) and only delay its printing.